gravitee-apim-cypress/cypress/integration/rest-api/apis/api-flow.ts

```diff
@@ -0,0 +1,145 @@
+      const fakeApi: Api = ApiFakers.api();
+      createApi(ADMIN_USER, fakeApi).should((response) => {
+        expect(response.body.state).equal('STOPPED');
+        expect(response.body.visibility).equal('PRIVATE');
+        expect(response.body.lifecycle_state).equal('CREATED');
+        api = response.body;
+      });
+    });
+
+    it('should fail to create an API if user lacks required permssions', function () {
+      const fakeApi: Api = ApiFakers.api();
+      createApi(LOW_PERMISSION_USER, fakeApi).should((response) => {
+        expect(response.status).to.equal(403);
+        expect(response.body.message).to.equal('You do not have sufficient rights to access this resource');
+      });
+    });
+  });
+
+  describe('Create a plan', () => {
+    it('should create a keyless plan as admin user', function () {
+      const fakePlan = PlanFakers.plan({ security: SecurityType.KEY_LESS });
+      createPlan(ADMIN_USER, api.id, fakePlan).should(function (response) {
+        expect(response.status).to.equal(201);
+        expect(response.body).to.have.all.keys(
+          'name',
+          'id',
+          'description',
+          'validation',
+          'security',
+          'type',
+          'status',
+          'api',
+          'order',
+          'created_at',
+          'updated_at',
+          'paths',
+          'flows',
+          'comment_required',
+        );
+        expect(response.body.security).to.equal('KEY_LESS');
+        expect(response.body.status).to.equal('STAGING');
+        plan = response.body;
+      });
+    });
+  });
+
+  describe('Publish plan', function () {
+    it('should publish a plan as admin user', function () {
+      publishPlan(ADMIN_USER, api.id, plan.id).should(function (response) {
+        expect(response.status).to.equal(200);
+        expect(response.body.status).to.equal('PUBLISHED');
+      });
+    });
+  });
+
+  describe('Publish an API', function () {
+    it('should publish an API as admin user', function () {
+      publishApi(ADMIN_USER, api).should(function (response) {
+        expect(response.status).to.equal(200);
+        expect(response.body.lifecycle_state).to.equal('PUBLISHED');
+        expect(response.body.state).to.equal('STOPPED');
+        expect(response.body.visibility).to.equal('PRIVATE');
+      });
+    });
+  });
+
+  describe('Start an API', function () {
+    it('should start an API as admin user', function () {
+      startApi(ADMIN_USER, api.id).should(function (response) {
+        expect(response.status).to.equal(204);
+      });
+    });
+  });
+
+  describe('Test newly created API', function () {
+    it('should get a positive response when calling the new API endpoint', 
+    {
+      "retries": 20
+    },
+    function () {
+      cy.request({
+        url: `${Cypress.env('gatewayServer')}${api.context_path}?teststring=${api.id}`,
+        retryOnStatusCodeFailure: true
+      }).its('body.query_params.teststring').should('equal', api.id);
```

Comment:
maybe we could add an assertion on status code too ?

Reply:
That's true, I'll add a check for status code 200.